goal_src/jak2/levels/forest/predator.gc

```diff
@@ -7,3 +7,1593 @@
+  ((predator-count    uint32                   :offset-assert 128)
+   (predator-h        handle                32 :offset-assert 136)
+   (hud-counter       handle                   :offset-assert 392)
+   (actor-group       (pointer actor-group)    :offset-assert 400)
+   (actor-group-count int32                    :offset-assert 404)
+   )
+  :heap-base #x120
+  :method-count-assert 21
+  :size-assert         #x198
+  :flag-assert         #x1501200198
+  (:methods
+    (idle () _type_ :state 14)
+    (closed () _type_ :state 15)
+    (predator-manager-method-16 (_type_) none 16)
+    (predator-manager-method-17 (_type_) none 17)
+    (predator-manager-method-18 (_type_) none 18)
+    (predator-manager-method-19 (_type_) none 19)
+    (predator-manager-method-20 (_type_) none 20)
+    )
+  )
+
+
+(define *predator-manager* (the-as predator-manager #f))
+
+;; WARN: Return type mismatch process vs predator-manager.
+(defmethod relocate predator-manager ((obj predator-manager) (arg0 int))
+  (set! *predator-manager* obj)
+  (if *predator-manager*
+      (set! *predator-manager* (&+ *predator-manager* arg0))
+      )
+  (the-as predator-manager ((method-of-type process relocate) obj arg0))
+  )
+
+(defmethod predator-manager-method-16 predator-manager ((obj predator-manager))
+  0
+  (none)
+  )
+
+(defstate idle (predator-manager)
+  :virtual #t
+  :enter (behavior ()
+    (set! (-> self hud-counter) (ppointer->handle (process-spawn hud-predator :init hud-init-by-other :to self)))
+    (none)
+    )
+  :exit (behavior ()
+    (when (-> self hud-counter)
+      (send-event (handle->process (-> self hud-counter)) 'hide-and-die)
+      (set! (-> self hud-counter) (the-as handle #f))
+      )
+    (none)
+    )
+  :trans (behavior ()
+    (predator-manager-method-16 self)
+    (let ((gp-0 0))
+      (dotimes (v1-2 (the-as int (-> self predator-count)))
+        (if (handle->process (-> self predator-h v1-2))
+            (+! gp-0 1)
+            )
+        )
+      (dotimes (s5-0 (length (-> self actor-group 0)))
+        (if (not (logtest? (-> self actor-group 0 data s5-0 actor extra perm status) (entity-perm-status dead)))
+            (+! gp-0 1)
+            )
+        )
+      (set! (-> *game-info* counter) (the float gp-0))
+      (when (zero? gp-0)
+        (task-node-close! (game-task-node forest-hunt-resolution))
+        (when (-> self hud-counter)
+          (send-event (handle->process (-> self hud-counter)) 'hide-and-die)
+          (set! (-> self hud-counter) (the-as handle #f))
+          (talker-spawn-func (-> *talker-speech* 89) *entity-pool* (target-pos 0) (the-as region #f))
+          )
+        )
+      )
+    (none)
+    )
+  :code (the-as (function none :behavior predator-manager) sleep-code)
+  )
+
+(defstate closed (predator-manager)
+  :virtual #t
+  :code (the-as (function none :behavior predator-manager) sleep-code)
+  )
+
+;; ERROR: function was not converted to expressions. Cannot decompile.
```

Comment:
maybe this is why they don't spawn?

Reply:
Yup, I overlooked some function that were marked as asm. They spawn now, but crash in merc code, likely related to their fancy cloak effect which uses generic.